docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -292,6 +304,45 @@ type LocalObjectTemplate struct {
   // offered by a provider.
   Ref *corev1.ObjectReference `json:"ref"`
 }
+
+// MachineHealthCheckClass defines a MachineHealthCheck for a group of Machines.
+type MachineHealthCheckClass struct {
+  // UnhealthyConditions contains a list of the conditions that determine
+  // whether a node is considered unhealthy. The conditions are combined in a
+  // logical OR, i.e. if any of the conditions is met, the node is unhealthy.
+  // Note: in order to provide a simpler UX, in case the user does not provide a value for this field, The topology reconciler applies
```

Comment:
Why do we "hide" the defaulting in the reconcile loop here instead of defaulting on the ClusterClass webhook? (like we afaik usually do) I think defaulting it in the ClusterClass webhook level would be more transparent for users (if we want to do it).

Also (applies to both variants) when we treat an empty `UnhealthyConditions` array like it has our default value, it seems to be impossible to configure a ClusterClass to not use `UnhealthyConditions`? Is that a problem?

Reply:
Just learned from Killian that the condition array is not allowed to be empty. So as an empty array is an invalid configuration my second point isn't a problem.

Reply:
The latest version no longer performs any defaulting on the MHC in the reconciler and leaves all defaulting the same as for vanilla MHC